.github/workflows/test.yml

```
@@ -0,0 +1,37 @@
+name: Tests
+
+on:
+  push:
+    paths-ignore:
+      - "*.md"
+  pull_request:
+    paths-ignore:
+      - "*.md"
+
+jobs:
+  tests:
+    name: Tests
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        include:
+        - golang: '^1.18'
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "${{ matrix.golang }}"
```

suggest: add cache for go modules https://github.com/actions/setup-go#caching-dependency-files-and-build-outputs
```suggestion
          go-version: "${{ matrix.golang }}"
          cache: true
```